src/components/BookstoreContainer.js

```diff
@@ -0,0 +1,66 @@
+import '../App.css';
+import React from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { Route, Routes } from 'react-router-dom';
+import Navbar from './Navbar';
+import BooksList from './BooksList';
+import AddBook from './AddBook';
+import Categories from '../pages/Categories';
+
+class BookStoreContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      books: [
+        {
+          id: uuidv4(),
+          category: 'Fiction',
+          title: 'Harry Potter and the Chamber of Secrets',
+          author: 'J.K. Rowling',
+          progress: 75,
+          currentChapter: 'Chapter 7',
+        },
+        {
+          id: uuidv4(),
+          category: 'Fiction',
+          title: 'The Hobbit',
+          author: 'J.R.R. Tolkien',
+          progress: 75,
+          currentChapter: 'Chapter 1',
+        },
+        {
+          id: uuidv4(),
+          category: 'Action',
+          title: 'The Hunger Games',
+          author: 'Suzanne Collins',
+          progress: 75,
+          currentChapter: 'Chapter 13',
+        },
+      ],
+    };
+  }
+
+  render() {
+    const allbooks = this.state;
+    return (
+      <>
+        <div className="App">
+          <Navbar />
+          <div className="container">
+            <Routes>
+              <Route path="/categories" element={<Categories />} />
+            </Routes>
+            <Routes>
+              <Route path="/book" element={<BooksList books={allbooks.books} />} />
+            </Routes>
+            <Routes>
+              <Route path="/book" element={<AddBook />} />
+            </Routes>
```

Comment:
- [ ] I would recommend removing the same path routes from here for the `AddBook` component and instead call the `AddBook` component from the `BookList` component. By this, it would maintain the hierarchy and the requirement as well.